app/src/main/java/org/fossasia/susi/ai/chat/STTfragment.kt

```diff
@@ -39,17 +44,42 @@ class STTFragment : Fragment() {
         val rootView = inflater.inflate(R.layout.fragment_sttframe, container, false)
         if (thisActivity is ChatActivity)
             thisActivity.fabsetting.hide()
-        promptSpeechInput()
+
+        textToSpeech = TextToSpeech(context, this)
+        val used_voice_state = PrefManager.getBoolean(R.string.used_voice, false)
+        if (used_voice_state == false) {
+            Handler().postDelayed({
+                speakOut()
+                PrefManager.putBoolean(R.string.used_voice, true)
+            }, 500)
```

Comment:
Why

Reply:
this is because we cannot start the speech engine as soon as the fragment on. The speech engine was failing to start. Upon, google search, I found that we should delay it by few moments to start the engine. After doing so, I found it worked.

Reply:
This is creating a memory leak

Reply:
I agree with what @iamareebjamal said. This is creating a memory leak because the anonymous runnable holds an implicit reference to your activity. Now, if your activity is destroyed within the delay duration after its creation, it would lead to a memory leak because the runnable will still hold a reference to the activity which will prevent the activity from being garbage collected.
Moreover, this code should not be put inside the fragment.

Reply:
@arundhati24 @iamareebjamal I will try to find a better solution and update the pr.